docs/_includes/examples/fakes/__tests.js

```diff
@@ -0,0 +1,185 @@
+"use strict";
+
+var proxyquire = require("proxyquire").noCallThru();
+var sinon = require("../../../../lib/sinon");
+var assert = require("@sinonjs/referee").assert;
+
+describe("`fake` examples", function () {
+    beforeEach(function () {
+        sinon.replace(global.console, "log", sinon.fake());
+    });
+
+    afterEach(function () {
+        sinon.restore();
+    });
+
+    // basic.js
+    describe("basic invocation", function () {
+        it("should output `1` on the console", function () {
+            proxyquire("./basic", {
+                "sinon": {
+                    fake: sinon.fake
+                }
+            });
+
+            sinon.assert.calledOnce(global.console.log);
+            sinon.assert.calledWithExactly(global.console.log, 1);
+        });
+    });
+
+    // returns.js
+    describe("returns", function () {
+        it("should output 'apple pie' on the console", function () {
+            proxyquire("./returns", {
+                "sinon": {
+                    fake: sinon.fake
+                }
+            });
+
+            sinon.assert.calledOnce(global.console.log);
+            sinon.assert.calledWithExactly(global.console.log, "apple pie");
+        });
+    });
+
+    // throws.js
+    describe("throws", function () {
+        it("should throw Error with message: 'not apple pie'", function () {
+            try {
+                proxyquire("./throws", {
```

Comment:
I didn't understand why `proxyquire` was necessary at first, ref comment above. Another thing is that I think all these proxyquire calls obscures what is going on. To be honest, if we intend to replace all instances of `sinon` in these tests with a link seam, I think this perfectly validates using the [`@global`](https://github.com/thlorenz/proxyquire#globally-override-require) flag to proxyquire. Using that, the tests would be simpler - most could be one-liners.

Example (this test):

```javascript
assert.exception(require.bind(null, "./throws"), "not apple pie");
```

Reply:
Maybe I am not reading the documentation right, but I am having a hard time seeing how the use of `@global` is going to turn the tests into one liners.

Could try to create a runnable example that demonstrates what you're thinking?

Reply:
Sorry - my comment doesn't make sense post-morning. Must have been high - on sleep depravation.